Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.swift

```diff
@@ -0,0 +1,227 @@
+                                    }
+                                }
+                            })
+                        }
+                    }
+                }
+                Section {
+                    AddAttachmentView(onAdd: {
+                        Task {
+                            do {
+                                guard let pngData = UIImage(named: "PinBlueStar")?.pngData()! else { return }
+                                try await model.addAttachment(
+                                    named: "Attachment",
+                                    type: "png",
+                                    dataElement: pngData
+                                )
+                            } catch {
+                                self.error = error
+                            }
+                        }
+                    })
+                }
+            }
+            .navigationTitle("Attachments")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        dismiss()
+                    }
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+// MARK: - CalloutView
+
+private extension EditFeatureAttachmentsView {
+    struct CalloutView: View {
+        /// The data model for the sample.
+        @ObservedObject var model: Model
+        
+        var body: some View {
+            VStack(alignment: .leading, spacing: 3) {
+                Text(model.calloutText)
+                    .font(.callout)
+                    .multilineTextAlignment(.leading)
+                Text(model.calloutDetailText)
+                    .font(.footnote)
+                    .multilineTextAlignment(.leading)
+            }
+        }
+    }
+}
+
+// MARK: - AttachmentView
+
+private extension EditFeatureAttachmentsView {
+    struct AttachmentView: View {
+        // The attachment that is being displayed.
+        let attachment: Attachment
+        // The closure called when the delete button is tapped.
+        let onDelete: ((Attachment) -> Void)
+        // The image in the attachment.
+        @State private var image: Image?
+        
+        var body: some View {
+            HStack {
+                Text(attachment.name)
+                    .font(.title3)
+                Spacer()
+                Button {
+                    Task {
+                        let result = try await attachment.data
+                        if let uiImage = UIImage(data: result) {
+                            image = Image(uiImage: uiImage)
+                        } else {
+                            image = Image(systemName: "exclamationmark.triangle")
+                        }
+                    }
+                } label: {
+                    if let image {
+                        image
```

Comment:
Would be better to apply some size constraints on the image. Currently if the image is huge, it would take up the entire row's space. I think for demo-ing purpose, fixed max/preferred width/height is good.

![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-15 at 16 27 15](https://github.com/user-attachments/assets/7ef0ab6e-bf68-4cf2-8e1a-fa50a0d43886)

Reply:
![Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 10 08 48](https://github.com/user-attachments/assets/61f3638d-65ad-47f9-9830-3f2e8ceadb3d)